harness/determined/experimental/pytorch_lightning/adapter.py

```diff
@@ -0,0 +1,79 @@
+import sys
+from inspect import signature
+from typing import Any, Callable, Dict, Sequence, Union
+
+import pytorch_lightning as ptl
+import torch
+
+from determined.pytorch import PyTorchTrial, PyTorchTrialContext
+
+TorchData = Union[Dict[str, torch.Tensor], Sequence[torch.Tensor], torch.Tensor]
+HyperparamsProvider = Callable[[str], Any]
+
+
+def bail(msg: str = "", fail: bool = True):
+    msg = f"NotSupported: {msg}"
+    if fail:
+        raise TypeError(msg)
+    else:
+        # QUESTION: do we have/want logger support.
+        print(msg, file=sys.stderr)
+
+
+def check_compat(lm: ptl.LightningModule):
+    if len(signature(lm.training_step).parameters) > 2:
+        bail("`optimizer_idx` and `hiddens` are not supported.")
+    if len(signature(lm.validation_step).parameters) > 2:
+        bail("`dataloader_idx` is not supported.")
+
+
+class PTLAdapter(PyTorchTrial):
```

Comment:
Will this class be instantiated by the trial controller directly? If so, then you shouldn't include `lightning_module` in the `__init__`. If it won't, then how you translate this class into a `PytorchTrial`?

Reply:
we discussed the usecase.